### PR TITLE
[MOB - 8993] - JWT part 2 with improvements

### PR DESCRIPTION
**Pull request by Ayyanchira**
## 🔹 Jira Ticket(s) if any

* [MOB-8993](https://iterable.atlassian.net/browse/MOB-8993)

## ✏️ Description

> Merges Feature branch JWT part 2 with fixes - with master

[MOB-8993]: https://iterable.atlassian.net/browse/MOB-8993?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableAuthManager.java [hunk 1]
```diff
@@ -125,8 +125,11 @@ private void handleAuthTokenSuccess(String authToken, IterableHelper.SuccessHand
         } else {
             IterableLogger.w(TAG, "Auth token received as null. Calling the handler in 10 seconds");
             //TODO: Make this time configurable and in sync with SDK initialization flow for auth null scenario
+            handleAuthFailure(authToken, AuthFailureReason.AUTH_TOKEN_NULL);
```
Comment (evantk91) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableAuthManager.java:128]: Should this TODO statement be removed?

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableAuthManager.java [hunk 2]
```diff
@@ -153,7 +156,8 @@ public void queueExpirationRefresh(String encodedJWT) {
             }
         } catch (Exception e) {
             IterableLogger.e(TAG, "Error while parsing JWT for the expiration", e);
-            authHandler.onTokenRegistrationFailed(new Throwable("Auth token decode failure. Scheduling auth token refresh in 10 seconds..."));
+            isLastAuthTokenValid = false;
+            handleAuthFailure(encodedJWT, AuthFailureReason.AUTH_TOKEN_PAYLOAD_INVALID);
             //TODO: Sync with configured time duration once feature is available.
```
Comment (evantk91) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableAuthManager.java:161]: Not sure if this TODO still needs to be done.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableRequestTask.java [hunk 3]
```diff
@@ -265,6 +266,40 @@ private static boolean matchesErrorCode(JSONObject jsonResponse, String errorCod
...
+            switch (errorMessage.toLowerCase()) {
+                case "exp must be less than 1 year from iat":
+                    return AuthFailureReason.AUTH_TOKEN_EXPIRATION_INVALID;
+                case "jwt format is invalid":
+                    return AuthFailureReason.AUTH_TOKEN_FORMAT_INVALID;
+                case "jwt token is expired":
+                    return AuthFailureReason.AUTH_TOKEN_EXPIRED;
+                case "jwt is invalid":
+                    return AuthFailureReason.AUTH_TOKEN_SIGNATURE_INVALID;
+                case "jwt payload requires a value for userid or email":
```
Comment (evantk91) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableRequestTask.java:286]: Is this missing a return statement?

**Review by evantk91 — COMMENTED** [review]
A few comments.

File: iterableapi/src/main/java/com/iterable/iterableapi/AuthFailure.java
[the remark this answers is not in the document]
Comment (Ayyanchira) [reply iterableapi/src/main/java/com/iterable/iterableapi/AuthFailure.java:1]: Which code section are you pointing this too?

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableRequestTask.java (hunk 3 above)
Comment (Ayyanchira) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableRequestTask.java:286]: No, this is how two conditions can be clubbed to return the same thing.

File: iterableapi/src/main/java/com/iterable/iterableapi/AuthFailure.java
[the remark this answers is not in the document]
Comment (Ayyanchira) [reply iterableapi/src/main/java/com/iterable/iterableapi/AuthFailure.java:1]: Got it

**Review by evantk91 — APPROVED** [review]

**Pull request merged**